estate/models/real_estate_offer.py

```diff
@@ -41,6 +41,23 @@ class RealEstateOffer(models.Model):
         ('check_price', 'CHECK(price>0)', 'The offer price must be stricly positive!')
     ]
 
+    @api.model
+    def create(self, vals):
+        if vals.get('property_id'):
+            property_id = self.env['real.estate'].browse(vals['property_id'])
```

Comment:
`browse()` must be suffixed by `exists()` because it is unreliable on its own. You can try to run `browse(123453759)` and obtain valid recordset containing invalid records. 
Alternatively you can use `serach(domain, limit=1)`
```suggestion
            property_id = self.env['real.estate'].browse(vals['property_id']).exists()
```